csharp/src/Ice/Internal/Instance.cs

```diff
@@ -1230,13 +1227,7 @@ public void destroy()
             Monitor.PulseAll(_mutex);
         }
 
-        {
-            Ice.FileLoggerI logger = _initData.logger as Ice.FileLoggerI;
-            if (logger != null)
-            {
-                logger.destroy();
-            }
-        }
+        _initData.logger?.Dispose();
```

Comment:
Do we own this logger, even if it i not a FileLoggerI? For example couldn't this be a logger from the user application that outlives the communicator?

Reply:
It's a good question, and it also applies to my Java Logger PR.

I think there are two options:

a) keep the old logic with try-to-cast to specific internal loggers before closing/disposing/destroying.
b) give ownership of the Logger to the communicator on set, via

https://github.com/zeroc-ice/ice/blob/7cdadd6b2bffa366ba4d795e4deedb71f1a4d5ef/csharp/src/Ice/LoggerPlugin.cs#L19 

or
https://github.com/zeroc-ice/ice/blob/7cdadd6b2bffa366ba4d795e4deedb71f1a4d5ef/csharp/src/Ice/Util.cs#L23

and of course, we need to release (dispose/close) any previous logger at this time.

Reply:
I was wondering if it would be clear to _not_ make the logger Disposable at all and remove destroy.

Is up to the implementations, in our case FileLoggerI can be Disposable, but the ConsoleLoggerI doesn't need to.

And then only dispose the logger if we create it and it is disposable.

Reply:
> if we create it and it is disposable.

This sounds complicated. For example, does the communicator "own" this cloned logger:
https://github.com/zeroc-ice/ice/blob/7cdadd6b2bffa366ba4d795e4deedb71f1a4d5ef/csharp/src/IceBoxNet/ServiceManagerI.cs#L543

Easier to switch back to (a).